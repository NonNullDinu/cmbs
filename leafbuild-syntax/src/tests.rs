fn parses_without_errors(s: &str) {
    assert_eq!(crate::parser::parse(s).errors, vec![]);
}

// Those are only *parsing* tests, so as long as they make sense
// *syntactically*, they should parse.
// Even if semantically they are void of meaning.
#[test]
fn unit_to_unit_assignment() {
    parses_without_errors("() = ()\n")
}

#[test]
fn line_trailing_space() {
    parses_without_errors("() = () \n")
}

#[test]
fn if_condition() {
    parses_without_errors("if (()) {} else {}")
}

#[test]
fn if_else_if_condition() {
    parses_without_errors("if () {} else if () {} else {}")
}

#[test]
fn if_else_if_else_condition_stretched() {
    parses_without_errors(
        r#"
    if ()

    {



    }

    else if () {


    }

    else {


    }
    "#,
    )
}

#[test]
fn if_condition_in_expr() {
    parses_without_errors("() = if () {} else {} \n")
}

#[test]
fn if_condition_in_expr_stretched() {
    parses_without_errors(
        r#"
    () = if ()
    {

    }

    else
    {


    }
    "#,
    )
}

#[test]
fn declaration() {
    parses_without_errors("let x = ()\n")
}

#[test]
fn foreach_basic() {
    parses_without_errors("foreach () in () {}")
}
