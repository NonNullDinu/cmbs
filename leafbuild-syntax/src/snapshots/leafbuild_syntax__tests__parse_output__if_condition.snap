---
source: leafbuild-syntax/src/tests.rs
expression: "if (()) {} else {}"

---
- ROOT 0..18
  - Conditional 0..18
    - ConditionalBranch 0..10
      - "if" IF_KW 0..2
      - " " WHITESPACE 2..3
      - Expr 3..7
        - PrimaryExpr 3..7
          - TupleExpr 3..7
            - "(" L_PAREN 3..4
            - Expr 4..6
              - PrimaryExpr 4..6
                - TupleExpr 4..6
                  - "(" L_PAREN 4..5
                  - ")" R_PAREN 5..6
            - ")" R_PAREN 6..7
      - ExprBlock 7..10
        - " " WHITESPACE 7..8
        - "{" L_BRACE 8..9
        - "}" R_BRACE 9..10
    - " " WHITESPACE 10..11
    - "else" ELSE_KW 11..15
    - " " WHITESPACE 15..16
    - ExprBlock 16..18
      - "{" L_BRACE 16..17
      - "}" R_BRACE 17..18

